src/main/java/ca/lambtoncollege/fsdm/s24/auction/repository/AuctionRepository.java

```diff
@@ -24,16 +23,20 @@ INSERT INTO Auction (title, description, min_bid, ends_at, image, state, created
             statement.setLong(3, auction.getMinBid());
             statement.setTimestamp(4, Timestamp.from(auction.getEndsAt()));
 
-            byte[] decodedByte = Base64.getDecoder().decode(PLACEHOLDER_IMAGE);
-            statement.setBlob(5, new SerialBlob(decodedByte));
+            if (auction.getImage() != null && !auction.getImage().isEmpty()) {
+                byte[] decodedImageBytes = Base64.getDecoder().decode(auction.getImage());
+                statement.setBlob(5, new SerialBlob(decodedImageBytes));
+            } else {
+                statement.setNull(5, java.sql.Types.BLOB);
+            }
             statement.setString(6, auction.getState().toString());
             statement.setInt(7, auction.getCreatedBy().getId());
 
             statement.executeUpdate();
             var keys = statement.getGeneratedKeys();
-            keys.next();
-
-            auction.setId(keys.getInt(1));
+            if (keys.next()) {
```

Comment:
Could we revert this back to what it was? If keys.next() does not return true, something is wrong and it should error out (there should always be an id)

Unless there's some edge case you found... let me know if you did